core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -336,11 +336,14 @@ public void setGameRule(MatchFinishEvent event, String gameRule, boolean gameRul
   /** Prevent teleporting outside the border */
   @EventHandler(priority = EventPriority.HIGH)
   public void onPlayerTeleport(final PlayerTeleportEvent event) {
-    if (event.getCause() == PlayerTeleportEvent.TeleportCause.PLUGIN) {
-      if (WorldBorders.isInsideBorder(event.getFrom())
-          && !WorldBorders.isInsideBorder(event.getTo())) {
-        event.setCancelled(true);
+    switch (event.getCause()) {
+      case PLUGIN -> {
+        if (WorldBorders.isInsideBorder(event.getFrom())
+            && !WorldBorders.isInsideBorder(event.getTo())) {
+          event.setCancelled(true);
+        }
       }
+      case NETHER_PORTAL, END_PORTAL -> event.setCancelled(true);
```

Comment:
If a map is set in the nether or end, will this prevent players being teleported to the match (ie: a normal cycle)?

Reply:
Well, I've checked the server code.

Modern handles this fine, as those causes are only used for the portal blocks.
For Sportpaper this will cause issues.

I suppose I should move this into `ModernListener`?